test/src/jest-xpaths.js

```diff
@@ -3,17 +3,27 @@ import 'expect-puppeteer';
 // Some extensions to the Jest expect API to better support XPaths
 expect.extend({
   async toContainXPath(page, xpath) {
-    return page.waitForXPath(xpath)
-      .then(() => ({ pass: true }));
+    try {
+      await page.waitForXPath(xpath);
+      return { pass: true }
```

Comment:
Do we want to support `expect(page).not.toContainXPath`? If so, [the API requires](https://jestjs.io/docs/en/expect#custom-matchers-api):

> And when `pass` is true, `message` should return the error message for when `expect(x).not.yourMatcher()` fails.

I don't have a preference.

Reply:
I was not aware of that pattern.
I think we should not support it for now. The behavior for `expect(page).not.toContainXPath` is very dependent on timeouts. If the timeout is too short, it will fail unexpectedly when the condition isn't met in time, but if it's too long, it can severely impact test runtime. Currently I think that timeout is globally configured to 30s.

I believe this will function as it currently is written, it just won't have a useful error message. If we run into a solid use case for using that pattern, it can be enabled easily, and we can tune the timeouts to something reasonable for that use case. Doing that now seems like premature optimization.

Reply:
Makes sense.